src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/index.ts

```diff
@@ -163,7 +163,15 @@
     datasetStats,
   }: MCDAConfig['layers'][0]) => {
     const [num, den] = axis;
-    const [min, max] = range;
+    let min = range[0];
+    const max = range[1];
+    // HACK: see #19471. Subtracting epsilon to avoid breaking MCDA calculation with 0 in denominator.
+    // TODO: Should apply proper solution later
+    if (min === max) {
+      // MapLibre expressions seem to have limited precision. So we need to apply Epsilon * 10 ^ number_of_digits
+      const digits = Math.trunc(min).toString().length;
+      min -= Number.EPSILON * Math.pow(10, digits);
+    }
```

Comment:
What is really needed here is nextafter function. 
Something like https://github.com/scijs/nextafter/tree/master maybe from other library?

@coderabbitai what's the best way to get nextafter or nextafterf in JS?

Reply:
@Komzpa Please check out updated solution. Utilized nextafter there, but had to make adjustments to the resulting value so that the difference doesn't get "eaten" by transformations